seumich/views.py

```diff
@@ -215,11 +217,14 @@ def get_class_history(self, student, class_site, format=None):
         except:
             return studentData, classData, activityData
 
-        events = class_site.weeklystudentclasssiteevent_set.filter(
-            student=student)
-        student_scores = class_site.weeklystudentclasssitescore_set.filter(
-            student=student)
-        class_scores = class_site.weeklyclasssitescore_set.all()
+        events = {x.get('week_end_date_id'):x
+            for x in class_site.weeklystudentclasssiteevent_set.filter(
+            student=student).values()}
+        student_scores = {x.get('week_end_date_id'):x
+            for x in class_site.weeklystudentclasssitescore_set.filter(
+            student=student).values()}
+        class_scores = {x.get('week_end_date_id'):x
+            for x in class_site.weeklyclasssitescore_set.all().values()}
```

Comment:
This is cool! It might make it a tad more readable if you used `event`, `student_score`, and `class_score` over `x`, but up to you.